### PR TITLE
Added the missing cluster resource options, docs and examples

### PR DESCRIPTION
**Pull request by KnockOutEZ**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,72 +22,301 @@ terraform {
...
+
+- **Add or remove nodes**: Modify the `nodes` block.
+- **Update regions and networks**: If you're adding or removing nodes, you must also update the corresponding `regions` and `networks` blocks for those nodes.
+- **Manage backup stores**: Adjust the `backup_store_ids` array if necessary.
+
+For example, when removing a node, ensure the corresponding region and network are also removed.
+
+---
+
+**Example: Removing a Node from the Cluster**
```
Comment (mmols) [inline README.md:183]: Update this example after the above changes.

File: README.md [hunk 2]
```diff
@@ -22,72 +22,301 @@ terraform {
...
     }
   ]
-}
 
-# Create a database
-resource "pgedge_database" "example" {
-  name       = "exampledb"
-  cluster_id = pgedge_cluster.main.id
+  backups = {
+    provider = "pgdump"
```
Comment (mmols) [inline README.md:260]: Let's show pgbackrest as the default example here for Enterprise, with the "standard" default config we provision in the UI (one config, daily full, hourly incremental)

File: docs/resources/backup_store.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+## Schema
+
+### Required
+
+- `cloud_account_id` (String)
+- `name` (String)
+
+### Optional
+
+- `region` (String)
```
Comment (mmols) [inline docs/resources/backup_store.md:25]: This field is required. Maybe something we need to fix in the schema

File: examples/resources/pgedge-cluster/resource.tf [hunk 4]
```diff
@@ -1,47 +1,51 @@
+resource "pgedge_cluster" "example" {
```
Comment (mmols) [inline examples/resources/pgedge-cluster/resource.tf:1]: Update these examples with the changes marked on the docs

Comment (mmols) [reply examples/resources/pgedge-cluster/resource.tf:1]: Alternatively, you can point to the examples from the docs if that makes sense?

**Review by mmols — COMMENTED** [review]
Here are some comments on the docs portions. Looking great overwall.

File: README.md [hunk 5]
```diff
@@ -22,72 +22,301 @@ terraform {
...
+terraform {
+  required_providers {
+    pgedge = {
+      source = "pgEdge/pgedge"
+    }
+  }
+}
+
+provider "pgedge" {
+  base_url = "https://api.pgedge.com"
```
[the remark this answers is not in the document]
Comment (KnockOutEZ) [reply README.md:79]: https://github.com/pgEdge/terraform-provider-pgedge/pull/8/files#diff-bd3a55a72186f59e2e63efb4951573b2f9e4a7cc98086e922b0859f8ccc1dd09R26-R57

https://github.com/pgEdge/terraform-provider-pgedge/pull/8/files#diff-ccc79b61f4ea5c957626ad87b0c415643fe4865cc42fe55c590e3aee4bd1396bR81


We do this for the client layer and provider layer now.

**Pull request merged**